include/boost/math/differentiation/autodiff_cpp11.hpp

```diff
@@ -91,7 +91,7 @@ template<typename... Orders>
 get_type_at<fvar<RealType,Order>,sizeof...(Orders)> fvar<RealType,Order>::derivative(Orders... orders) const
 {
   static_assert(sizeof...(Orders) <= depth, "Number of parameters to derivative(...) cannot exceed fvar::depth.");
-  return at(orders...) * product(boost::math::factorial<root_type>(orders)...);
+  return at(static_cast<std::size_t>(orders)...) * product(boost::math::factorial<root_type>(static_cast<unsigned>(orders))...);
```

Comment:
Why this change?

Reply:
I made the types that `fvar::at` and `boost::math::factorial` explicit (both MSVC and Clang complain about `orders` not being `unsigned` for `boost::math::factorial` specifically).

Reply:
Thanks. That makes sense for `factorial`. Is the `static_cast<std::size_t>(orders)` necessary to satisfy a warning as well?